src/server/lib/idapi/IDAPICookies.ts

```diff
@@ -37,9 +37,22 @@ export const setIDAPICookies = (
       }
     })
     .forEach(({ key, value, sessionCookie = false }) => {
+      const expires: Date | undefined = (() => {
+        if (key === 'SC_GU_LA') {
+          // have SC_GU_LA cookie expire in 30 mins, despite the fact that it is a session cookie
+          // this is to support the Native App logging in with an in-app browser, so the cookie
+          // isn't immediately deleted when the in app browser is closed
+          return new Date(Date.now() + 30 * 60 * 1000);
+        }
+        if (sessionCookie) {
+          return undefined;
+        }
+        return new Date(expiresAt);
+      })();
```

Comment:
used an IIFE instead of two ternary operators as I find this easier to read

Reply:
With ternary operators this would become

```suggestion
      // have SC_GU_LA cookie expire in 30 mins, despite the fact that it is a session cookie
      // this is to support the Native App logging in with an in-app browser, so the cookie
      // isn't immediately deleted when the in app browser is closed
      const expires: Date | undefined =
        key === 'SC_GU_LA'
          ? new Date(Date.now() + 30 * 60 * 1000)
          : sessionCookie
          ? undefined
          : new Date(expiresAt);
```

Reply:
which i personally find difficult to understand, but open to opinions

Reply:
I suppose you could make it a regular function, but I (grudgingly) agree that an IIFE is better than a ternary!

Reply:
Yeah i made it an IIFE as it's the only place it's used, so thought we didn't need to make a separate function just for this behaviour.